spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverJsonLayout.java

```diff
@@ -163,6 +165,17 @@ public void setIncludeExceptionInMessage(boolean includeExceptionInMessage) {
 		this.includeExceptionInMessage = includeExceptionInMessage;
 	}
 
+	@Override
+	public void start() {
+		super.start();
+
+		// If no Project ID set, then attempt to resolve it with the default project ID provider
+		if (StringUtils.isEmpty(this.projectId) || this.projectId.endsWith("_IS_UNDEFINED")) {
+			GcpProjectIdProvider projectIdProvider = new DefaultGcpProjectIdProvider();
```

Comment:
I'm not sure about using the `DefaultGcpProjectIdProvider` here. If we want to be consistent with the rest of the project, we should add `projectId` property to `StackdriverLoggingProperties`.

Reply:
I wish there is a way to get that info from the Layout. I haven't found a way though.

Reply:
You're right. The problem is that it lives outside of the Spring Context and initialized by Logback.